File: DataBindingHandlerJava/app/src/main/java/com/websarva/wings/android/databindinghandlerjava/MainActivity.java
package com.websarva.wings.android.databindinghandlerjava;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;
import androidx.lifecycle.ViewModelProvider;

import android.os.Bundle;

import com.websarva.wings.android.databindinghandlerjava.databinding.ActivityMainBinding;
import com.websarva.wings.android.databindinghandlerjava.ui.MainHandlers;
import com.websarva.wings.android.databindinghandlerjava.ui.MainViewModel;

public class MainActivity extends AppCompatActivity {
	@Override
	protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		ActivityMainBinding activityMainBinding = DataBindingUtil.setContentView(MainActivity.this, R.layout.activity_main);
		ViewModelProvider provider = new ViewModelProvider(MainActivity.this);
		MainViewModel mainViewModel = provider.get(MainViewModel.class);
		MainHandlers mainHandlers = new MainHandlers(mainViewModel, activityMainBinding);
		activityMainBinding.setRectangle(mainViewModel.getRectangle());
		activityMainBinding.setMainHandlers(mainHandlers);
	}
}

File: DataBindingHandlerKotlin/app/src/main/java/com/websarva/wings/android/databindinghandlerkotlin/MainActivity.kt
package com.websarva.wings.android.databindinghandlerkotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.activity.viewModels
import androidx.databinding.DataBindingUtil
import com.websarva.wings.android.databindinghandlerkotlin.databinding.ActivityMainBinding
import com.websarva.wings.android.databindinghandlerkotlin.ui.MainHandlers
import com.websarva.wings.android.databindinghandlerkotlin.ui.MainViewModel

class MainActivity : AppCompatActivity() {
	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)
		val activityMainBinding = DataBindingUtil.setContentView<ActivityMainBinding>(this@MainActivity, R.layout.activity_main)
		val mainViewModel by viewModels<MainViewModel>()
		val mainHandlers = MainHandlers(mainViewModel, activityMainBinding)
		activityMainBinding.rectangle = mainViewModel.rectangle
		activityMainBinding.mainHandlers = mainHandlers
	}
}
